src/main/java/qupath/ext/wsinfer/models/WSInferUtils.java

```diff
@@ -104,15 +83,52 @@ public static WSInferModelCollection downloadModelCollection() {
         return cachedModelCollection;
     }
 
+    /**
+     * Check if a directory exists and create it if it does not.
+     * @param path the path of the directory
+     * @return true if the directory exists when the method returns
+     */
+    public static boolean checkPathExists(Path path) {
+        if (!Files.exists(path)) {
+            try {
+                Files.createDirectories(path);
+            } catch (IOException e) {
+                logger.error("Cannot create directory {}", path, e);
+                return false;
+            }
+        }
+        return true;
+    }
+
+    private static File getCachedCollectionFile() {
```

Comment:
Not a big deal, but it strikes me that you could probably just consistently use `Path` instead of `File`, and then you wouldn't need to keep converting with `File.toPath()`.

Construction could also be easier with [`Paths.get(String first, String more...)`](https://docs.oracle.com/en/java/javase/20/docs/api/java.base/java/nio/file/Paths.html#get(java.lang.String,java.lang.String...)) and you wouldn't have to explicitly add in the separators.

Reply:
I keep trying to use one over the other, but there's an annoying mix of `File`s and `Path`s needed by different methods

Reply:
Yeah, I have the impression that I *should* use `Path` since it's more modern and recommended, but then keep finding cases where a `File` is needed.

(FYI it seems that `Files.exists` can have a nasty performance impact if used a lot... at least it seemed so [here](https://github.com/qupath/qupath/issues/1154#issuecomment-1327044045) and a search for 'Java 8's "Files.exists" should not be used' returns some other hits... but I do still use it for isolated checks)

Reply:
Remarkable

The solution? `myPath.toFile().exists()`, apparently